app/src/main/java/com/douglasbruce/tasky/features/event/EventViewModel.kt

```diff
@@ -254,17 +259,88 @@ class EventViewModel @Inject constructor(
                                 state = state.copy(closeScreen = true)
                             }
 
-                            is AuthResult.Unauthorized -> {
-                                state.copy(logout = true)
+                            else -> {}
+                        }
+                    }
+                }
+            }
+
+            is EventFormEvent.ToggleAddVisitorDialogClick -> {
+                state = state.copy(showAddVisitorDialog = event.show)
+            }
+
+            is EventFormEvent.AttendeeEmailValueChanged -> {
+                val result = emailValidator.validate(event.email)
+                state = state.copy(
+                    attendeeEmail = event.email,
+                    attendeeEmailErrorType = result.errorType,
+                    isAttendeeEmailValid = result.successful
+                )
+            }
+
+            is EventFormEvent.OnAddAttendeeClick -> {
+                state = state.copy(isCheckingAttendee = true)
+                val eventId = state.id ?: UUID.randomUUID().toString()
+
+                viewModelScope.launch {
+                    when (val result = eventRepository.getAttendee(state.attendeeEmail)) {
+                        is AuthResult.Success -> {
+                            val remindAt = NotificationType.notificationTypeToZonedDateTime(
+                                state.fromDate,
+                                state.fromTime,
+                                state.notificationType
+                            )
+
+                            result.data?.let { networkAttendee ->
+                                if (networkAttendee.doesUserExist) {
+                                    networkAttendee.attendee?.let { attendee ->
+                                        val newAttendee = Attendee(
+                                            email = attendee.email,
+                                            fullName = attendee.fullName,
+                                            userId = attendee.userId,
+                                            isGoing = true,
+                                            eventId = eventId,
+                                            remindAt = remindAt.toInstant().toEpochMilli(),
+                                        )
+
+                                        state = state.copy(
+                                            id = eventId,
+                                            showAddVisitorDialog = false,
+                                            attendeeEmail = "",
+                                            isAttendeeEmailValid = false,
+                                            attendeeEmailErrorType = ErrorType.NONE,
+                                            attendees = state.attendees.plus(newAttendee),
```

Comment:
I personally find the normal plus operator more readable: `state.attendees + newAttendee`